liquidity-mining/src/lib.rs

```diff
@@ -0,0 +1,1416 @@
+        OptionQuery,
+    >;
+
+    /// Deposit details.
+    #[pallet::storage]
+    #[pallet::getter(fn deposit)]
+    pub type Deposit<T: Config<I>, I: 'static = ()> =
+        StorageMap<_, Blake2_128Concat, DepositId, DepositData<T, I>, OptionQuery>;
+
+    /// Active(farms able to receive LP shares deposits) yield farms.
+    #[pallet::storage]
+    #[pallet::getter(fn active_yield_farm)]
+    pub type ActiveYieldFarm<T: Config<I>, I: 'static = ()> =
+        StorageDoubleMap<_, Blake2_128Concat, T::AmmPoolId, Blake2_128Concat, GlobalFarmId, YieldFarmId>;
+
+    #[pallet::call]
+    impl<T: Config<I>, I: 'static> Pallet<T, I> {}
+}
+
+impl<T: Config<I>, I: 'static> Pallet<T, I> {
+    /// Create new liquidity mining program with provided parameters.
+    ///
+    /// `owner` account have to have at least `total_rewards` balance. This funds will be
+    /// transferred from `owner` to farm account.
+    ///
+    /// Returns: `(GlobalFarmId, max reward per period)`
+    ///
+    /// Parameters:
+    /// - `total_rewards`: total rewards planned to distribute. This rewards will be
+    /// distributed between all yield farms in the global farm.
+    /// - `planned_yielding_periods`: planned number of periods to distribute `total_rewards`.
+    /// WARN: THIS IS NOT HARD DEADLINE. Not all rewards have to be distributed in
+    /// `planned_yielding_periods`. Rewards are distributed based on the situation in the yield
+    /// farm and can be distributed in a longer never in the shorter time frame.
+    /// - `blocks_per_period`:  number of blocks in a single period. Min. number of blocks per
+    /// period is 1.
+    /// - `incentivized_asset`: asset to be incentivized in AMM pools. All yield farms added into
+    /// global farm must to have `incentivized_asset` in their pair.
+    /// - `reward_currency`: payoff currency of rewards.
+    /// - `owner`: liq. mining farm owner.
+    /// - `yield_per_period`: percentage return on `reward_currency` of all pools
+    #[allow(clippy::too_many_arguments)]
+    pub fn create_global_farm(
+        total_rewards: Balance,
+        planned_yielding_periods: PeriodOf<T>,
+        blocks_per_period: BlockNumberFor<T>,
+        incentivized_asset: AssetIdOf<T, I>,
+        reward_currency: AssetIdOf<T, I>,
+        owner: AccountIdOf<T>,
+        yield_per_period: Permill,
+        min_deposit: Balance,
+    ) -> Result<(GlobalFarmId, Balance), DispatchError> {
+        Self::validate_create_global_farm_data(
+            total_rewards,
+            planned_yielding_periods,
+            blocks_per_period,
+            yield_per_period,
+            min_deposit,
+        )?;
+
+        ensure!(
+            T::MultiCurrency::free_balance(reward_currency, &owner) >= total_rewards,
+            Error::<T, I>::InsufficientRewardCurrencyBalance
+        );
+
+        let planned_periods =
+            TryInto::<u128>::try_into(planned_yielding_periods).map_err(|_e| ArithmeticError::Overflow)?;
```

Comment:
Also, i would probably choose different error - there is already something like InvalidPeriod. 

```suggestion
            TryInto::<u128>::try_into(planned_yielding_periods).map_err(|_| ArithmeticError::Overflow)?;
```